rllib/algorithms/sac/torch/sac_torch_rl_module.py

```diff
@@ -0,0 +1,134 @@
+from typing import Any, Dict, List, Tuple
+
+from ray.rllib.algorithms.sac.sac_rl_module import QF_PREDS
+from ray.rllib.algorithms.sac.sac_rl_module import SACRLModule
+from ray.rllib.core.models.base import ENCODER_OUT
+from ray.rllib.core.rl_module.torch.torch_rl_module import TorchRLModule
+from ray.rllib.core.rl_module.rl_module import RLModule
+from ray.rllib.core.rl_module.rl_module_with_target_networks_interface import (
+    RLModuleWithTargetNetworksInterface,
+)
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.utils.typing import NetworkType
+
+
+torch, nn = try_import_torch()
+
+
+class SACTorchRLModule(TorchRLModule, SACRLModule):
+    framework: str = "torch"
+
+    @override(RLModule)
+    def _forward_inference(self, batch: NestedDict) -> Dict[str, Any]:
+        output = {}
+
+        # Pi encoder forward pass.
+        pi_encoder_outs = self.pi_encoder(batch)
+
+        # Pi head.
+        output[SampleBatch.ACTION_DIST_INPUTS] = self.pi(pi_encoder_outs[ENCODER_OUT])
+
+        return output
+
+    @override(RLModule)
+    def _forward_exploration(self, batch: NestedDict) -> Dict[str, Any]:
+        return self._forward_inference(batch)
+
+    @override(RLModule)
+    def _forward_train(self, batch: NestedDict) -> Dict[str, Any]:
+        output = {}
+
+        # SAC needs also Q function values and action logits for next observations.
+        # TODO (simon): Check, if we need to override the Encoder input_sp
+        batch_curr = NestedDict({SampleBatch.OBS: batch[SampleBatch.OBS]})
```

Comment:
Dumb question: Would it work all w/o `NestedDict` (use simple dicts instead)? Not a huge fan of these as they add additional burden on the user's mental model.

Reply:
It expects a `NestedDict` in the `RLModule` `forward methods.

Reply:
It works without `NestedDict`

Reply:
```suggestion
        batch_curr = {SampleBatch.OBS: batch[SampleBatch.OBS]}
```